riyadh-h/markbind
# Hide site nav under a button

**What is the purpose of this pull request?**




**Overview of changes:**
Hides the site nav section under a button on the navbar, making it easy to show/hide the site nav section when necessary.

The site nav section might not be needed at all times, especially within long, full-of-text pages.

Therefore, hiding the site nav section would give more display space to the contents of the site's pages.

Before:
![site-nav-before](https://user-images.githubusercontent.com/69278589/129193590-9e2de489-b0a2-449c-a2ac-e7dcc07bcf02.png)


After (hidden):
![site-nav-after-hidden](https://user-images.githubusercontent.com/69278589/129193606-1658b601-d353-4e71-84ae-2ca5e5d21065.png)

After (shown):
![site-nav-after-shown](https://user-images.githubusercontent.com/69278589/129193654-e4ff1f58-c557-4880-ad03-5389ace129f8.png)

**Anything you'd like to highlight / discuss:**
* This change is implemented as a MarkBind plugin, allowing for easier enabling/disabling of the enhancement through the `plugins` property of `site.json`. The plugin is enabled by default.

* The show/hide site nav button adopts the theme of the site's navbar. Examples of this are shown below:
![site-nav-theme-dark](https://user-images.githubusercontent.com/69278589/129196539-b1cb108c-bd93-41d9-be11-350da6dade55.png)
![site-nav-theme-primary](https://user-images.githubusercontent.com/69278589/129196541-db9d417a-67d2-4117-a4ae-ee4e62b45048.png)
![site-nav-theme-light](https://user-images.githubusercontent.com/69278589/129196542-c823bf05-5f39-40f2-8376-baec2f50a286.png)


* This change is open for further development (e.g., changing the position of the show/hide button).

**Testing instructions:**
For new MarkBind instances, initialize and serve such instances. The changes should appear automatically.

For existing MarkBind instances...
* If the `plugins` property exists, add `"hiddenSiteNav"` to the `plugins` array.
* If the `pluginsContext` property exists, add the following to the `pluginsContext` object:
```
"hiddenSiteNav": {
    "navbarTheme": "dark"
}
```
*Note: the `dark` value for `navbarTheme` acts as a placeholder.*
* If neither `plugins` nor `pluginsContext` exist, add the following to the `site.json` object:
```
"plugins": [
    "hiddenSiteNav"
],
"pluginsContext": {
    "hiddenSiteNav": {
        "navbarTheme": "dark"
    }
}
```


**Proposed commit message: (wrap lines at 72 characters)**
> Pages display the site nav section when exceeding a certain window width.
> 
> The site nav section usually takes unnecessary space within the site's pages.
> 
> Hiding the site nav section allows us to increase the display space of our pages' contents.
> 
> Let's hide the site nav section behind a button on the navbar.
> 
> Using a button makes it easy to show/hide the site nav section with a single click.


---

**Checklist:** :ballot_box_with_check:
